Added the rank method under ModulesWithBasis.ParentMethod.

HenryWu2101 opened the pull request:
#34445
It is based on the understanding that dimension is equal rank under the context, assuming rank defined for infinite dimensional space.

<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For example, "Fixes #12345". -->



### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example, -->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

HenryWu2101 committed:
Added the rank method under ModulesWithBasis.ParentMethod.

It is based on the understanding that dimension is equal rank under the context,
assuming rank defined for infinite dimensional space.

tscrim commented on src/sage/categories/modules_with_basis.py:
```diff
@@ -1308,7 +1308,29 @@ def dimension(self):
+            """
+            Return the rank of ``self``.
+            Note the rank of ``self`` should equal dimension of ``self``,
+            assuming infinite rank is defined in the context.
```
```suggestion
        def rank(self):
            """
            Return the rank of ``self``.

            Since there is a (distinguished) basis, the rank of ``self``
            is equal to the cardinality of the basis (which equals
            the :meth:`dimension` of ``self``).
```

HenryWu2101 replied:
Hi thanks for your feedback!
I've modified the suggested doc tweak.

tscrim reviewed (changes requested):
Thank you for doing this PR; sorry it took so long for me to look at it.

Just a little doc tweak suggestion. Otherwise LGTM.

HenryWu2101 committed:
Modified the suggested doc tweak.

tscrim reviewed (approved):
Thank you. Once the bots come back (morally) green, I will set a positive review.

HenryWu2101 committed:
Corrected the coding style violation

This commit only changes coding style violation in the comment. No code is altered.
